src/main/java/googlesignin/GoogleUser.java

```diff
@@ -67,27 +48,41 @@ public String getPhoneNumber(String userId) throws IOException {
                 .execute();
         List<PhoneNumber> phoneNumberList = profile.getPhoneNumbers();
         if(phoneNumberList.isEmpty()) {
-            return null;
+            return Optional.empty();
         }
-        return phoneNumberList.get(0).getValue();
+        return Optional.of(phoneNumberList.get(0).getValue());
     }
 
-    public User getUser() throws IOException {
+    /**
+     * Retrieves the username and email-id of signed-in user
+     * Checks if an entry with the same username and email-id exists
+     * If it exists, then return the UserID of existing user
+     * Else create a new entry in the User table
+     * And return the UserID of the newly created user
+     */
+    public long getOrCreateUser() throws IOException {
         GoogleIdToken idToken = tokenResponse.parseIdToken();
         GoogleIdToken.Payload payload = idToken.getPayload();
         String email = payload.getEmail();
-        String username = googleUserMapper.getUsernameFromEmail(email);
-        User existingUser = userAccessor.checkIfUserExists(username, email);
-        if(existingUser != null) {
-            return existingUser;
+        String username = GoogleUserMapper.getUsernameFromEmail(email);
+        Optional<Long> existingUser = userAccessor.getUserIdFromUsernameAndEmail(username, email);
+        if(existingUser.isPresent()) {
```

Comment:
What if a user exists with the same email but not same username, or same username but not same email?
In that case, `existingUser` will be empty as the function checks for `AND` and we will proceed to create a new user with either a conflicting email or a conflicting username.

Please fix this bug.

Reply:
Done.